### ISSUE TITLE
Group leaders cannot subscribe to hidden feeds on screens not owned by their group

### ISSUE THREAD

**Issue by ddbruce** [concerto/concerto#1333]

**Comment by augustf** [comment]
Just tested this out on the demo instance, and this behavior is by design. That is, a feed that isn't visible cannot be subscribed to by someone who doesn't have the proper permissions to the feed and the screen.

**Comment by ddbruce** [comment]
I created the feed and added the content, then tried to add the feed to a screen that I also created. In this instance, I still could not add the feed. Is this by design?

**Comment by augustf** [comment]
You'll want to make sure that you do own both the screen and the feed (or have full permissions). If you like, try reproducing your issue on the demo instance (http://www.concerto-signage.org/demo) so we can make sure we're seeing the same issue.

**Comment by ddbruce** [comment]
On the demo site, using "Katie"'s administrator credentials, I set up Test Screen 1 and Test Feed 1, the latter being non-submittable and non-visible. Here's a screenshot of me trying to add the feed to the screen after adding content to the feed.

![image](https://cloud.githubusercontent.com/assets/5694673/13988658/54b8e79a-f0e3-11e5-948c-e088cbe5015f.png)

**Comment by augustf** [comment]
Well that would be expected, right? You're the owner of the screen, feed, and control the group that owns them. When I login as mary@concerto-signage.org (pwd: marymac), I'm unable to see the Concerto feed, either for submitting content or subscribing to.

![screen shot 2016-03-23 at 2 12 17 pm](https://cloud.githubusercontent.com/assets/12512/13995830/59fb7a42-f101-11e5-8fef-607fd79b83f2.png)

**Comment by ddbruce** [comment]
How would that be expected? I would expect that Katie (especially as an administrator) should be able to add a secret and private feed to a screen of her choice, especially when Katie owns all of those things. I understand if Mary can't see the feed, but Katie owns both the screen and the feed.

Use case: a private screen in the Office of the President at RPI. They want to use the same Concerto instance because they want the "Service and Community" feed, but they also want graphics that are private and only for the use of that office. They create a non-submittable and non-viewable feed on Concerto, then add it to a screen that they administer. Currently, they cannot add the feed to their screen.

**Comment by augustf** [comment]
Ahh - I see what you mean. We were using different screens and feeds to test this. What I didn't notice about your screen was that it was owned by Katie, not Concerto Admins. When I changed the ownership to Concerto Admins, things worked as expected (see below). So it's true that if the same user or the same group own the feed and the screen, things work as expected. But if one is owned by the group leader, and the other by the group proper, we see the results you're getting.

![screen shot 2016-03-23 at 11 11 46 pm](https://cloud.githubusercontent.com/assets/12512/14007425/f52f574a-f14c-11e5-99ca-3ea28f48efe4.png)

I believe the way forward here is to make the rule for private feeds a bit "fuzzier" to include group leaders. I'll update and rename this issue to make that clearer.

**Comment by gbprz** [comment]
I just logged into Katie's account on the demo instance and noticed she was never a member of the Concerto Admins group (even though she was a system wide admin). I added her to the group and promoted her to leader. 

I then tested this out using a screen Katie owns (test2) and the hidden feed she created (Test feed 1). 

While you may have created the feed on her account, it seems like feeds are only owned by groups and not users (no user id is saved to indicate they own the feed). Even though Katie owned the screen and created the feed, she was not a member of the group that owned the hidden feed. Once she was added to Concerto Admins and promoted to leader (the group that moderates your hidden feed Test feed 1), the subscription list now includes Test feed 1.

I also tested just giving Katie all permissions for feeds and screens instead of promoting her to leader of Concerto Admins. This also appears to work and I can add the hidden feed.

TL;DR If a user owns a screen and created the feed **but isn't a member of the group that owns the hidden feed**, you can't add the hidden feed because there is no way to tie the feed's owner (a group) to the screen's owner (a user). 

I just removed her from the group again so someone else can try this out for themselves. Let me know if I'm misunderstanding something here.

**Comment by augustf** [comment]
Yeah-I thought there was something odd here, but Dan and I both failed to notice it.
